test/test.c

```diff
@@ -29485,6 +29500,9 @@ _jsOrderedErrHandler(natsConnection *nc, natsSubscription *subscription, natsSta
 {
     struct threadArg    *args = (struct threadArg*) closure;
 
+    if (err != NATS_MISSED_HEARTBEAT)
```

Comment:
There were other errors coming in. At the end of the first cycle, there was a `NATS_ERR`, I didn't track it down although it seemed a bit suspiciously generic.

After the end of the test, the callback was getting called (twice?) more with `5` (connection closed), and was locking up the Mutex being destroyed in the main thread already. Ignoring other errors fixed the warning, not sure if the termination is clean.